hugegraph-api/src/main/java/com/baidu/hugegraph/api/graph/VertexAPI.java

```diff
@@ -259,22 +313,53 @@ private static void checkBatchSize(HugeConfig config,
         int max = config.get(ServerOptions.MAX_VERTICES_PER_BATCH);
         if (vertices.size() > max) {
             throw new IllegalArgumentException(String.format(
-                      "Too many counts of vertices for one time post, " +
+                      "Too many vertices for one time post, " +
                       "the maximum number is '%s'", max));
         }
     }
 
-    @JsonIgnoreProperties(value = {"type"})
-    private static class JsonVertex implements Checkable {
+    private Id getVertexId(HugeGraph g, String labelId, JsonVertex vertex) {
+        assert g.vertexLabel(labelId).idStrategy() != IdStrategy.AUTOMATIC;
+
+        List<Id> pkIds = g.vertexLabel(vertex.label).primaryKeys();
+        List<Object> pkValues = new ArrayList<>(pkIds.size());
+        for (Id pkId : pkIds) {
+            String propertyKey = g.propertyKey(pkId).name();
+            Object propertyValue = vertex.properties.get(propertyKey);
+            E.checkState(propertyValue != null,
+                         "The value of primary key '%s' can't be null",
+                         propertyKey);
+            pkValues.add(propertyValue);
+        }
+
+        String name = SplicingIdGenerator.concatValues(pkValues);
+        Object idValue = vertex.id != null ? vertex.id :
+                         SplicingIdGenerator.splicing(labelId, name);
+        return HugeVertex.getIdValue(idValue);
+    }
+
+    private static class VertexRequest {
+
+        @JsonProperty("jsonVertices")
```

Comment:
RESTful API params use underlined style, EdgeAPI is an exception, in order to follow the specifications of tinkerpop
@JsonProperty("jsonVertices") -> @JsonProperty("vertices")
@JsonProperty("updateStrategies") -> @JsonProperty("update_strategies")
@JsonProperty("create") -> @JsonProperty("create_if_not_exist")